.github/workflows/continous-integration.yml

```diff
@@ -39,15 +39,20 @@ jobs:
     - name: Install dependencies 🖥
       run: poetry install --no-interaction
 
-    - name: Lint Code 🎎
-      run: make lint
```

Comment:
Can this line not just be replaced with the following?
```yaml
- uses: pre-commit/action@v2.0.3
```

Then the `pre-commit` job can be removed.

Reply:
since we already utilize make in multiple places. Then let's try to go with make for `pre-commit`.
All libs should be already set, so make should do the work

Reply:
In that case, can `make lint` just be changed to `make formatter`? Do we need the additional `pre-commit` job?

Reply:
The `make lint` command used to display a diff of any changes instead of changing the files. Does `pre-commit` have an option for this?

Reply:
and `flake` will show the error lines and error code. So, only `black` is not verbose, we can modify it by adding verbose flag, but nobody does it in `pre-commit`, fixing `black` style manually is not a good time investment